### PR TITLE
Bat-native-ledger linter fixes (part 2)

### PR DESCRIPTION
**Pull request by NejcZdovc**
Partial for https://github.com/brave/brave-browser/issues/3413

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- check if it builds and do general rewards tests

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/include/bat/ledger/publisher_info.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 namespace ledger {
 
-const std::string clear_favicon = "clear";
+const char clear_favicon[] = "clear";
```
Comment (tmancey) [inline vendor/bat-native-ledger/include/bat/ledger/publisher_info.h]: nit: globals should start with _, i.e. `_clear_favicon`

File: vendor/bat-native-ledger/src/bat/ledger/ledger.cc [hunk 2]
```diff
@@ -63,7 +63,7 @@ bool VisitData::loadFromJson(const std::string& json) {
 
   // has parser errors or wrong types
   bool error = d.HasParseError();
-  if (false == error) {
+  if (!error) {
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/ledger.cc]: nit: code would be easier to read as follows:
```
bool success = !d.HasParseError();
if (!success) {
  // Fail...
}
```

File: vendor/bat-native-ledger/src/bat_client.cc [hunk 3]
```diff
@@ -91,8 +96,8 @@ void BatClient::requestCredentialsCallback(
   std::vector<uint8_t> publicKey;
   std::vector<uint8_t> newSecretKey;
   braveledger_bat_helper::getPublicKeyFromSeed(secretKey,
-                                               publicKey,
-                                               newSecretKey);
+                                               &publicKey,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat_client.cc]: Linter does not pick these up yet, but we should not be using camel-case in C++, i.e. `newSecretKey` and should be using snake-case, i.e. `new_secret_key`

File: vendor/bat-native-ledger/src/bat/ledger/ledger.cc (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/ledger.cc]: agree, but wanted to minimise changes

File: vendor/bat-native-ledger/src/bat_client.cc (hunk 3 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat_client.cc]: agree, but let's do this is a separate PR

File: vendor/bat-native-ledger/include/bat/ledger/publisher_info.h (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/include/bat/ledger/publisher_info.h]: done

**Pull request merged**